idf_build_apps/manifest/manifest.py

```diff
@@ -68,11 +81,43 @@ def _clause_to_if_clause(clause: t.Dict[str, t.Any]) -> IfClause:
                 _kwargs['reason'] = clause['reason']
             return IfClause(**_kwargs)
 
+        def _clauses_to_switch_clause(statements: t.Optional[t.List[t.Union[str, t.Dict[str, t.Any]]]]) -> SwitchClause:
+            if not statements:
+                return SwitchClause([], [], [])
+
+            switch_statements = []
+            no_switch_statements = []
+            for statement in statements:
+                if isinstance(statement, t.Dict):
+                    switch_statements.append(statement)
+                else:
+                    no_switch_statements.append(statement)
+
+            if switch_statements and no_switch_statements:
+                raise InvalidManifest('Current manifest format has to fit either the switch format or the list format.')
+
+            ifs = []
+            values = []
+            default = []
+            for statement in switch_statements:
+                if 'if' in statement:
+                    ifs.append(IfClause(stmt=statement['if']))
+                    values.append(statement['content'])
+                elif 'default' in statement:
+                    default.extend(statement['default'])
+                else:
+                    raise InvalidManifest("Only the 'if' and 'default' keywords are supported in switch clause.")
+
+            for statement in no_switch_statements:
+                default.append(statement)
+
+            return SwitchClause(ifs, values, default)
+
         self.enable = [_clause_to_if_clause(clause) for clause in enable] if enable else []
         self.disable = [_clause_to_if_clause(clause) for clause in disable] if disable else []
         self.disable_test = [_clause_to_if_clause(clause) for clause in disable_test] if disable_test else []
-        self.depends_components = depends_components or []
-        self.depends_filepatterns = depends_filepatterns or []
+        self.depends_components = _clauses_to_switch_clause(depends_components)
+        self.depends_filepatterns = _clauses_to_switch_clause(depends_filepatterns)
```

Comment:
I'm not sure if it's a breaking change or not.

I recommend to keep the original data type. i.e.

- list of dictionary -> `SwitchClause`
- list of string -> `list[str]`
- None -> `[]`

Reply:
I added a few more lines to the code, but typing right now